src/utils/hooks/useSubmitApplication.tsx

```diff
@@ -18,6 +18,25 @@ const useSubmitApplication = ({ serialNumber }: UseGetApplicationProps) => {
     },
   })
 
+  const submitFromStructure = async (structure: FullStructure) => {
+    const elements = Object.values(structure.elementsById || {}).filter(
+      (element) => element.element.category === TemplateElementCategory.Question
+    )
+    const responsesPatch = elements.map(({ latestApplicationResponse }) => {
+      return {
+        id: latestApplicationResponse.id,
+        patch: { value: latestApplicationResponse.value || null },
+      }
+    })
+
+    return await applicationSubmitMutation({
+      variables: {
+        serial: serialNumber,
+        responses: responsesPatch,
+      },
+    })
+  }
+
```

Comment:
Cool. Do we need to keep the previous `submit` method at all? Can we get rid of it?

Reply:
ebac6961fd8a9e0cb1bda196272054f4f1138ffe

added a comment to remove it (I think it's still needed in the old application)